stateMutators.js

```diff
@@ -0,0 +1,111 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const moveBasketLeftKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos - value,
+});
+
+const moveBasketRightKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos + value,
+});
+
+const moveBasketMouse = (state, value) => ({
```

Comment:
This mutator is exactly the same as `setBasketValue` below. And the name indicates it's for the mouse movement, which is something for later I suppose. So this prob can be removed

Reply:
How is it that same as setBasketValue? moveBasketMouse has the basketPos property and setBasketValue has the basketValue property? Am I maybe not understanding you properly?

Reply:
You're right. I thought they were both mutating `basketPos`, but they're different in that way. Too tired when I looked at this :S 

Though yeah I'd still rename this `setBasketPos` to remove the tight coupling to the idea of the context in which this mutator will be used (similar to the `moveBasketRightKey` one below)

Reply:
So something like `moveBasket` and then remove `moveBasketLeft`, `moveBasketRight` and `moveBasketMouse`?

Reply:
no, the `moveBasketLeft` and `moveBasketRight` names are fine. They correspond to core changes in the game's state/how the game works. 

`moveBasketMouse` does not. This name corresponds to a specific input method, which makes it hard to be used in other contexts because the name ties it to a single specific way of use. So i'd just rename this `setBasketPosition` then it can be used for the mouse movement and any other context in which we need to move the basket later.

Naming is a surprisingly difficult part of programming in my experience. It's easy to write some names that makes sense at one time due to all the other stuff you have going through your head, though then give it to other people, or to your future self/s, who aren't thinking the same and the meaning gets interpreted differently leading to miscommunications. Thinking about things in different layers of responsiblity can help with this (e.g., main UI -> user input handling -> data (state) changes)
Some more thoughts on naming: https://hilton.org.uk/blog/why-naming-things-is-hard

Reply:
Thanks Jon, that makes sense. So seeing as I already have a mutator called setBasketPos, I removed the moveBasketMouse function all together